parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDMLStatementVisitor.java

```diff
@@ -669,6 +669,23 @@ public ASTNode visitParenthesisSelectSubquery(final ParenthesisSelectSubqueryCon
         return visit(ctx.selectSubquery());
     }
     
+    private void setSelectCombineClause(final SelectCombineClauseContext ctx, final OracleSelectStatement result, final OracleSelectStatement left) {
+        for (int i = 0; i < ctx.selectSubquery().size(); i++) {
+            CombineType combineType;
+            if (null != ctx.UNION(i) && null != ctx.ALL(i)) {
+                combineType = CombineType.UNION_ALL;
+            } else if (null != ctx.UNION(i)) {
+                combineType = CombineType.UNION;
+            } else if (null != ctx.INTERSECT(i)) {
+                combineType = CombineType.INTERSECT;
+            } else {
+                combineType = CombineType.MINUS;
+            }
+            result.setCombine(new CombineSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex(), left,
```

Comment:
Why call setCombine in foreach? I think combine should be a multi-level nested structure.